matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/mapper/EventAnnotationsSummaryMapper.kt

```diff
@@ -36,13 +37,22 @@ internal object EventAnnotationsSummaryMapper {
                     )
                 },
                 editSummary = annotationsSummary.editSummary
-                        ?.let {
-                            val latestEdition = it.editions.maxByOrNull { editionOfEvent -> editionOfEvent.timestamp } ?: return@let null
+                        ?.let { summary ->
+                            /**
+                             * The most recent event is determined by comparing origin_server_ts;
+                             * if two or more replacement events have identical origin_server_ts,
+                             * the event with the lexicographically largest event_id is treated as more recent.
+                             */
+                            val latestEdition = summary.editions.sortedWith(compareBy<EditionOfEvent> { it.timestamp }.thenBy { it.eventId })
+                                    .lastOrNull() ?: return@let null
+                            // get the event and validate?
```

Comment:
Why is there this comment? Is there a validation step missing?

Reply:
no should be validated before, or for late decryption after